### PR TITLE
API Explorer: Header search

### PR DESCRIPTION
**Pull request by josephaxisa**
Replaced the sidebar search with a header search to adhere to design spec. 
- Used `Combobox` and `ComboboxInput` for the search bar
- Search criteria not accounted for in the design doc were added under a gear icon next to the search bar for now
- Made search disabled when no search criteria are selected
- Search bar is focused on slash key press
- Search placeholder shows the criteria where the inputted search pattern will be searched for

TODO: 

- [x] - Handle invalid/erroneous search patterns
- [x] - Tests

Demo:
search demo

### CODE REVIEW

File: packages/api-explorer/src/components/Search/SearchError.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  error: string
+}
+
+export const SearchError: FC<SearchErrorProps> = ({ error }) => (
+  <ComboboxList>
+    <ListItem>
+      <Icon name="Error" size="small" color="critical" mr="xxsmall" />
+      <Text>{error}</Text>
+    </ListItem>
+  </ComboboxList>
```
Comment (jkaster) [inline packages/api-explorer/src/components/Search/SearchError.tsx:39]: does this mean handling search errors is done?

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ export const SideNavMethods: FC<MethodsProps> = ({ methods, tag, specKey }) => {
             <SidebarItem key={method.name} as="span">
               <>
                 <MethodBadge verb={method.httpMethod} />
-                {highlightHTML(pattern, method.operationId)}
+                {highlightHTML(pattern, method.summary)}
```
Comment (jkaster) [inline packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:54]: Are we going to do a truncate here, or are you waiting until switching to `Accordion`?

File: packages/api-explorer/src/components/Search/SearchError.tsx (hunk 1 above)
Comment (josephaxisa) [reply packages/api-explorer/src/components/Search/SearchError.tsx:39]: indeed!

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx (hunk 2 above)
Comment (josephaxisa) [reply packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:54]: I'll get that in here. No reason to wait.

**Pull request merged**